backend/service/topology/topology.go

```diff
@@ -187,7 +188,11 @@ func (c *client) Search(ctx context.Context, req *topologyv1.SearchRequest) ([]*
 	return results, nextPageTokenStr, nil
 }
 
-func (c *client) Autocomplete(ctx context.Context, typeURL, search string, limit uint64) ([]*topologyv1.Resource, error) {
+func (c *client) Autocomplete(ctx context.Context, typeURL, search string, limit uint64, caseInsensitive bool) ([]*topologyv1.Resource, error) {
+	if caseInsensitive {
+		search = strings.ToLower(search)
```

Comment:
i'm just curious how this works without being passed down to the actual search function. haven't been in the loop on this bug fix, so maybe it's already been documented somewhere else.

Reply:
but the database just happens to be lowercase at Lyft, is that right? couldn't it be mixed case when dealing with data from other sources?

Reply:
here was an alternative PR https://github.com/lyft/clutch/pull/2256 that just lowercased the FE. But that was decided to be the wrong approach after a discussion.

Reply:
I thought about this more - lets have 2 modes, fast and slow. Fast just lowercases the string. Slow adds the `LIKE` keyword with collation. So instead of `caseInsensitive` being a bool it could be an enum instead. Does this plan sound acceptable?

Reply:
i wouldn't overcomplicate it. is the performance of ILIKE really unacceptable for our data set?

alternatively the postgres docs suggest just using `lower()` in the query https://www.postgresql.org/docs/current/indexes-expressional.html, which you can also create an index on if performance is still not acceptable. i would go that direction i think.